Delicious/Scenes/Home/HomeViewModel.swift

```diff
@@ -14,14 +14,67 @@ struct HomeViewModel {
 // MARK: - ViewModelType
 extension HomeViewModel: ViewModelType {
     struct Input {
-
+        let loadTrigger: Driver<Void>
+        let reloadTrigger: Driver<Void>
+        let selectTrigger: Driver<RecipeType>
+        let searchTrigger: Driver<Void>
     }
 
     struct Output {
-
+        let data: Driver<[HomeTableViewSection]>
+        let isLoading: Driver<Bool>
+        let isReloading: Driver<Bool>
+        let isEmpty: Driver<Bool>
+        let placeHolderError: Driver<Error>
+        let loadedError: Driver<Error>
+        let selected: Driver<Void>
+        let search: Driver<Void>
     }
 
-    func transform(_ input: Input) -> Output {
-        return Output()
+    func transform(_ input: HomeViewModel.Input) -> HomeViewModel.Output {
+        let error = ErrorTracker()
+
+        let getRecipes = getItem(
+            loadTrigger: input.loadTrigger,
+            reloadTrigger: input.reloadTrigger) { _ in
+            return self.useCase.getRecipes().trackError(error)
+        }
+
+        let recipe = getRecipes.item.map { recipes -> [HomeTableViewSection] in
+            let data = recipes.split()
+            let featuredSection = HomeTableViewSection.featuredSection(
+                item: .featuredItem(items: data.left)
+            )
+            let lastestSection = HomeTableViewSection.latestSection(
+                items: data.right.map { HomeTableViewItem.latestItem(item: $0) }
+            )
+            return [featuredSection, lastestSection]
+        }
+
+        let isEmpty = recipe.map { $0.isEmpty }
+
+        let isPlaceholderEnabled = Driver.merge(Driver<Bool>.of(true), isEmpty).distinctUntilChanged()
+
+        let placeHolderError = error.withLatestFrom(isPlaceholderEnabled) { $1 ? $0: nil }.compactMap { $0 }
+
+        let loadedError = error.withLatestFrom(isEmpty) { $1 ? $0: nil }.compactMap { $0 }
+
+        let selected = input.selectTrigger.do(onNext: { recipe in
```

Comment:
nên xuống break line cho đẹp mắt hơn

input.selectTrigger
.do(onNext: { recipe in

}